protocols/shuffling+ddt_protocol.go

```diff
@@ -152,8 +153,19 @@ func (p *ShufflingPlusDDTProtocol) Start() error {
 func (p *ShufflingPlusDDTProtocol) Dispatch() error {
 	defer p.Done()
 
-	shufflingPlusDDTBytesMessageLength := <-p.LengthNodeChannel
-	tmp := <-p.PreviousNodeInPathChannel
+	var shufflingPlusDDTBytesMessageLength shufflingPlusDDTBytesLengthStruct
+	select {
+	case shufflingPlusDDTBytesMessageLength = <-p.LengthNodeChannel:
+	case <-time.After(libunlynx.TIMEOUT):
+		return errors.New(p.ServerIdentity().String() + " didn't get the <shufflingPlusDDTBytesMessageLength> on time.")
+	}
+
+	var tmp shufflingPlusDDTBytesStruct
+	select {
+	case tmp = <-p.PreviousNodeInPathChannel:
+	case <-time.After(libunlynx.TIMEOUT):
+		return errors.New(p.ServerIdentity().String() + " didn't get the <tmp> on time.")
```

Comment:
Maybe a better name than `tmp`?

Reply:
I tried to clean a bit these `tmp` variables. I changed this one to `spDDTbs` (referring to the type of the object `shufflingPlusDDTBytesStruct`) even though is not that much better.

a4d36eb